### ISSUE TITLE
Add more important assertions to twisted/trial/test/test_tests.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5859]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5859|
|Type|enhancement|
|Created|2012-08-13 13:43:07Z|
|Branch|https://github.com/twisted/twisted/tree/extra-reporter-assertions-5859|

Many of the tests in `test_tests.py` operate by running a suite and then making assertions on the results/reporter object.  Some of these tests skip checking certain parts of the reporter's state, though, which potentially allows a broken implementation to pass them.

The tests should have a few more assertions added to make sure a properly working implementation is really in use.

This is split off of [#5853](https://github.com/twisted/twisted/issues/5853).

<details><summary>Searchable metadata</summary>

```
trac-id__5859 5859
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_extra_reporter_assertions_5859 branches/extra-reporter-assertions-5859
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1344865387000000 1344865387000000
changetime__1344884077000000 1344884077000000
version__None None
owner__exarkun exarkun
cc__jml
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [35170]) Branching to 'extra-reporter-assertions-5859'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [35171]) Add a few extra assertions to some existing test methods.  Also add missing docstrings, cleanup whitespace, and fix test names.

refs [#5859](https://github.com/twisted/twisted/issues/5859)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun removed owner|
|-|-|

Okay, made those improvements.  There's surely more that _could_ be done here, like unifying the logic for asserting a particular result, rather than repeating the same (or, more often than not, not _quite_ the same) group of 5 or 8 individual assertions.  But I added the extra coverage I wanted, at least, and made some other minor cleanups as well.

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/extra-reporter-assertions-5859)

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)|@Julian set owner to @Julian|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)<a name="note_6"></a>|@Julian set owner to @exarkun|
|-|-|

Hello :).

There's a typo in `TestSkipClasses`'s `test_results` test method (a doubled period). Other than that, what's here looks fine :).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [35176]) Merge extra-reporter-assertions-5859

Author: exarkun
Reviewer: Julian
Fixes: [#5859](https://github.com/twisted/twisted/issues/5859)

Expand the test coverage for what is done to the test results object passed in to
a trial test suite.  Also, clean up a few other minor issues such as whitespace and
docstrings.
